modules/chapter1/pages/section1.adoc

```diff
@@ -18,27 +18,38 @@ Good machine learning scenarios often have the following common properties:
 
 Reinforcement learning:: 
-It involves an agent learning to make decisions through trial and error interactions with an environment. The agent receives feedback in the form of rewards or penalties for its actions, guiding it towards achieving a goal by maximizing the rewards. E.g Autonomous driving, AlphaGo learning to play Go by playing against itself.
+In comparison to the previous groups of algorithms mentioned, reinforcement learning does not have the goal of grouping or classifying data. It tries to find the most optimal path. It involves an agent learning to make decisions through trial and error interactions with an environment. The agent receives feedback in the form of rewards or penalties for its actions, guiding it towards achieving a goal by maximizing the rewards. E.g Autonomous driving, AlphaGo learning to play Go by playing against itself. Examples of algorithms are Monte Carlo, Deep Q Networks(DQN), State-Action-Reward-State-Acction(SARSA) and others. 
 
+== Choosing your algorithm
+We have a plethora of mathematical algorithms that may, or may not, solve our problem and be applied on our data. But, how do you recognise the category of problem and the algorithm that can help you solve it?
+The clue lies with your outcome variable.
+ 
+In a classification problem, the outcome variable (the event you want to predict) is a category or class label. It can be binary, such as "Sick" or "Healthy", "Yes" or "No". They can be multi-class such as for image recognition when you want to identify more than 2 classes or labels. 
+
+In a regression problem, the outcome variable is a continuous value and the goal is predicting a quantitative value. Common examples are predicting house prices, temperatures and sale revenues. 
+
+Knowing the outcome variable you are able to identify if its supervised, unsupervised, semi-supervised or reinforcement learning. However, there is a lot of work left before the final algorithm is chosen. The data scientist will spend time exploring the data set, applying different visualization and data cleaning technqiues exploring the underlying data, and several different machine learning algorithms may be applied to test which one solves the problem. Solving the problem is multifacetted and there is no one algorithm to rule them all.  
+ 
 == What is a Machine Learning model
 
-A Machine Learning model is a file that has been trained to recognize certain types of patterns. You train a model over a set of data, providing it with an algorithm that it can use to reason over and learn from that data.
+A Machine Learning model is a file that has been trained to recognize certain types of patterns. You train a mdel over a set of data, providing it with an algorithm that it can use to reason over and learn from that data.
```

Comment:
```suggestion
A Machine Learning model is a file that has been trained to recognize certain types of patterns. You train a model over a set of data, providing it with an algorithm that it can use to reason over and learn from that data.
```